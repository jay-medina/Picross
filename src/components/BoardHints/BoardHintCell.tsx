import React from 'react';

export interface BoardHintCellProps {
  hint: number;
}

const BoardHintCell: React.StatelessComponent<BoardHintCellProps> = ({ hint }) => (
  <div className="cell hint">
    {hint}
  </div>
);

export default BoardHintCell;
