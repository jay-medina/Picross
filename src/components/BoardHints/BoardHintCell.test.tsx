import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import BoardHintCell from './BoardHintCell';

describe('<BoardHintCell />', () => {
  let wrapper: ShallowWrapper;

  describe('when the component renders', () => {
    beforeEach(() => {
      wrapper = shallow(
        <BoardHintCell hint={4} />,
      );
    });

    it('displays the hint in the cell', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });
});
