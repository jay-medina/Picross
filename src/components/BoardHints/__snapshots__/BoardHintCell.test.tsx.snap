// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardHintCell /> when the component renders displays the hint in the cell 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardHintCell
    hint={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": 4,
      "className": "cell hint",
    },
    "ref": null,
    "rendered": 4,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 4,
        "className": "cell hint",
      },
      "ref": null,
      "rendered": 4,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
